module/zcommon/zfeature_common.c

```diff
@@ -381,6 +381,7 @@ zpool_feature_init(void)
 		SPA_FEATURE_EXTENSIBLE_DATASET,
 		SPA_FEATURE_NONE
 	};
+	/* NOTE: Use org.openzfs (without the dash) for new feature GUIDs. */
```

Comment:
What if instead of annotating each one of these instances of `org.zfsonlinux` we added a comment above the `zpool_feature_init()` function which briefly summarized the policy.  Here's a first attempt, which could still use a bit more editing.

```c
/*
 * Each feature is prefixed by a reversed domain name so that it can be
 * uniquely identified.  This has allowed companies to independently develop
 * features without first reserving the feature name with the OpenZFS
 * project (org.delphix, org.datto).  Similarly, features developed on one
 * platform have used the platform's domain name (org.zfsonlinux, org.illumos).
 * Use of the "org.openzfs" domain is recommended for new features which
 * are developed by the community.  The domain may optionally be used by
 * companies or platforms provided they reserve the feature name in
 * advance with the OpenZFS project developers.
 */
```

Reply:
I spent some time editing on this. In the course of doing so, I came to the conclusion that where/when the features are going to be released is a significant factor. If a company (like Delphix or Datto) wants to be able to release their feature before it lands in OpenZFS, then they should use their own domain name. If they use the OpenZFS domain name, then the on-disk format for that feature is already committed, which ties the hands of the OpenZFS project. It seems that the OpenZFS domain should only be used when the code is going to be initially released through an OpenZFS platform.

The first sentence is from the man page. The rest is from what you wrote, with edits from me.
```
/*
 * Every feature has a GUID of the form com.example:feature_name.  The
 * reversed DNS name ensures that the feature's GUID is unique across all ZFS
 * implementations.  This allows companies to independently develop and
 * release features.  Examples include org.delphix and org.datto.  Previously,
 * features developed on one implementation have used that implementation's
 * domain name (e.g. org.illumos and org.zfsonlinux).  Use of the org.openzfs
 * domain name is recommended for new features which are developed by the
 * OpenZFS community and its platforms.  This domain may optionally be used by
 * companies developing features for initial release through an OpenZFS
 * implementation.  Use of the org.openzfs domain requires reserving the
 * feature name in advance with the OpenZFS project.
 */
```